lightningd/lightningd.c

```diff
@@ -263,6 +274,57 @@ static const char *subdaemons[] = {
 	"lightning_openingd"
 };
 
+/* Return true if called with a recognized subdaemon */
+bool is_subdaemon(const char *sdname)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(subdaemons); i++)
+		if (strcmp(sdname, subdaemons[i]) == 0)
+			return true;
+	return false;
+}
+
+static void destroy_alt_subdaemons(struct lightningd *ld)
+{
+	strmap_clear(&ld->alt_subdaemons);
+}
+
+#if DEVELOPER
+static void memleak_help_alt_subdaemons(struct htable *memtable,
+					struct lightningd *ld)
+{
+	memleak_remove_strmap(memtable, &ld->alt_subdaemons);
+}
+#endif /* DEVELOPER */
+
+const char *subdaemon_path(const struct lightningd *ld, const char *name)
+{
+	const char *dpath;
+
+	/*~ CCAN's path module uses tal, so wants a context to
+	 * allocate from.  We have a magic convenience context
+	 * `tmpctx` for temporary allocations like this.
+	 *
+	 * Because all our daemons at their core are of form `while
+	 * (!stopped) handle_events();` (an event loop pattern), we
+	 * can free `tmpctx` in that top-level loop after each event
+	 * is handled.
+	 */
+
+	/* Is there an alternate path for this subdaemon? */
+	const char *alt = strmap_get(&ld->alt_subdaemons, name);
+	if (alt) {
+		/* Is the alternate path absolute? */
+		if (alt[0] == '/')
```

Comment:
`path_is_abs(alt)` is already defined in ccan ;-)

Reply:
fixed